xskillscore/core/deterministic.py

```diff
@@ -6,14 +6,61 @@
 __all__ = ['pearson_r', 'pearson_r_p_value', 'rmse', 'mse', 'mae']
 
 
-def _preprocess(dim):
+def _preprocess_dims(dim):
+    """Preprocesses dimensions to prep for stacking.
+    
+    Parameters
+    ----------
+    dim : str, list
+        The dimension(s) to apply the function along.
+    """
     if isinstance(dim, str):
         dim = [dim]
     axis = tuple(range(-1, -len(dim) - 1, -1))
     return dim, axis
 
 
-def pearson_r(a, b, dim):
+def _preprocess_weights(a, dim, new_dim, weights):
+    """Preprocesses weights array to prepare for numpy computation.
+    
+    Parameters
+    ----------
+    a : Dataset, DataArray, GroupBy, Variable, numpy/dask arrays or scalars
+        One of the arrays over which the function will be applied.
+    dim : str, list
+        The original dimension(s) to apply the function along.
+    new_dim : str
+        The newly named dimension after running ``_preprocess_dims``
+    weights : Dataset, DataArray, GroupBy, Variable, numpy/dask arrays or scalars
+        Weights to apply to function, matching the dimension size of ``new_dim``.
+    """
+    if weights is None:
+        return xr.full_like(a, None)  # Return nan weighting array.
```

Comment:
This along with L52 forces input arrays to be DataArray or Dataset. Since this is an `xarray`-based package that relies on calling a `dim` by label, is it reasonable to enforce that the input data (a, b) are `xarray` objects? [Following https://github.com/raybellwaves/xskillscore/issues/29]. L52 would be hard to pull off without dimension labeling.

In other words, this breaks the ability to input ndarrays.